assets/src/components/detours/detourModal.tsx

```diff
@@ -0,0 +1,39 @@
+import React, { useEffect } from "react"
+import { DiversionPage } from "./diversionPage"
+import { joinClasses } from "../../helpers/dom"
+import { OriginalRoute } from "../../detour"
+
+export const DetourModal = ({
+  originalRoute,
+  onClose,
+  show,
+}: {
+  originalRoute: OriginalRoute
+  onClose: () => void
+  show: boolean
+}) => {
+  useEffect(() => {
+    const handleEscape = (event: KeyboardEvent) => {
+      if (event.key === "Escape") {
+        onClose()
+      }
+    }
+
+    window.addEventListener("keydown", handleEscape)
```

Comment:
**question:** do we necessarily want this event listener on `window`? Does it work all right if we just attach it to the top-level `<div>` of the modal?

Reply:
I know that with having this event listener on `window` we don't need to manage focus because _something_ will hear it, but I agree that it'd be nicer to have this event in the actual tree of the component, and manage focus instead.

Fine with this being non-blocking though.

Reply:
It is kinda weird that when I open the modal and press tab nothing happens because the focus is still on the map page (god I hate modals >.>)

Reply:
I created two follow-up tickets:

 - [⚙️ Have the detour modal grab focus when it renders](https://app.asana.com/0/1148853526253420/1206648035312400/f)
 - [⚙️ Move the escape-key handler onto the detour modal instead of the whole window](https://app.asana.com/0/1152340551558956/1206648035312404/f)